### PR TITLE
[WIP] Upgrade to DyNet

### PR DESCRIPTION
**Pull request by neubig**
This PR should update the stack LSTM NER to work with the most recent version of DyNet. @adhigunasurya is still testing whether the accuracy/speed are comparable, so please wait to merge until we confirm this.

### CODE REVIEW

**Comment by Jaaaaaay** [comment]
thank your contribution, I want to use your programmer to solve Chinese text for NER, An error occurred, Does this programmer  not fit Chinese context or something else?

**Comment by miguelballesteros** [comment]
Should work in any language in which you have data

**Comment by Jaaaaaay** [comment]
ok, my training data is like conll2003, but when the training started, this error information "[1]  + 22834 segmentation fault   -T  -d  --hidden_dim 100 --lstm_input_dim 100 -w  --pretrained_dim 100  20 " occurred. How to fix this error?

**Comment by miguelballesteros** [comment]
You should give more information, try harder and then open open your own issue. Don't write in what Graham did to update the code to a newer version of Dynet 

Thanks
